### PR TITLE
Structured Query Language definition

### PR DESCRIPTION
**Pull request by SerekKiri**
This Pull Request adds Structured Query Language definition.

### CODE REVIEW

File: data/Language/SQL.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  - UNIQUE
+  - UPDATE
+  - VALUES
+  - VIEW
+  - WHERE
+datatypes:
+  - bool_title_case
+  - float_sql
+  - int
+  # TODO  character, varchar, smallint, decimal, date, numeric, real, time, timestamp, clob, blob, double precision
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: comment not indented like content (comments-indentation)

*Origin: YAMLLintBear, Section: `yaml`.*

Comment (jayvdb) [inline data/Language/SQL.yaml]: line too long (115 > 80 characters) (line-length)

*Origin: YAMLLintBear, Section: `yaml`.*

Comment (jayvdb) [inline data/Language/SQL.yaml]: no new line character at the end of file (new-line-at-end-of-file)

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/SQL.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  - UNIQUE
+  - UPDATE
+  - VALUES
+  - VIEW
+  - WHERE
+datatypes:
+  - bool_title_case
+  - float_sql
+  - int
+  # TODO  character, varchar, smallint, decimal, date,
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: comment not indented like content (comments-indentation)

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/SQL.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  # numeric, real, time, timestamp, clob, blob, double precision
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: no new line character at the end of file (new-line-at-end-of-file)

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/SQL.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+  - UNIQUE
+  - UPDATE
+  - VALUES
+  - VIEW
+  - WHERE
+datatypes:
+  - bool_title_case
+  - float_sql
+  - int
+    # TODO  character, varchar, smallint, decimal, date,
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: comment not indented like content (comments-indentation)

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/SQL.yaml [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  - smallint         # TODO
+  - decimal          # TODO
+  - date             # TODO
+  - numeric          # TODO
+  - real             # TODO
+  - time             # TODO
+  - timestamp        # TODO
+  - clob             # TODO
+  - blob             # TODO
+  - double precision # TODO
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: too few spaces before comment (comments)

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/DataType/float_sql.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+keyword: float
+integer-min: '-12345678'
+integer-max: '123.45678'
```
Comment (siddhpant) [inline data/DataType/float_sql.yaml]: Is this intended??

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by jayvdb** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by jayvdb** [comment]
Comment on 6fa5f385d16da701323280a3b2b8ae439c7a072e, file data/Language/VisualBasic.yaml, line 5.

line too long (82 > 80 characters) (line-length)

*Origin: YAMLLintBear, Section: `yaml`.*

File: data/Language/SQL.yaml [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+full_name: Structured Query Language
+extensions:
+  - sql
+delimiters:
+  # comment delimters:
+  - double_dash
+  - multiline_slash_star
+keywords:
+  - ADD
+  - ADD CONSTRAINT
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: DDL

File: data/Language/SQL.yaml [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+  - SELECT INTO
+  - SELECT TOP
+  - SET
+  - TABLE
+  - TOP
+  - TRUNCATE TABLE
+  - UNION
+  - UNION ALL
+  - UNIQUE
+  - UPDATE
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: DML

File: data/Language/SQL.yaml [hunk 9]
```diff
@@ -0,0 +1,105 @@
+identifier: SQL
+wikidata: Q47607
+full_name: Structured Query Language
```
Comment (ayan-b) [inline data/Language/SQL.yaml]: Please add `creation_date` and `versions`.

File: data/Language/SQL.yaml [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+  - TRUNCATE TABLE
+  - UNION
+  - UNION ALL
+  - UNIQUE
+  - UPDATE
+  - VALUES
+  - VIEW
+  - WHERE
+datatypes:
+  - bool_title_case
```
Comment (jayvdb) [inline data/Language/SQL.yaml]: this includes `keyword: bool`.  is that appropriate for SQL?

And SQL is case insensitive.

So the `literals:` also needs to be case insensitive in order to be correct.  Good luck with that.

File: data/DataType/float_sql.yaml (hunk 6 above)
Comment (jayvdb) [reply data/DataType/float_sql.yaml]: Which spec is this in?

**Comment by SerekKiri** [comment]
Task abandoned

**Pull request closed without merging**